nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBControllerService.java

```diff
@@ -111,41 +111,57 @@ protected MongoClient createClient(ConfigurationContext context, MongoClient exi
             final String authMechanism;
             if (authMechanismMatcher.find()) {
                 authMechanism = authMechanismMatcher.group(1);
-                uri = authMechanismMatcher.replaceFirst(uri.contains("?") ? "?" : "");
-                // If trailing & or ? is left, clean up
-                uri = uri.replaceAll("[&?]+$", "");
             } else {
                 authMechanism = null;
             }
 
-            final ConnectionString cs = new ConnectionString(uri);
+            // When properties specify the user, and the URI includes an authMechanism but no user in the URI,
+            // the Mongo driver would attempt to build credentials from the URI and fail. In that case, remove the
+            // authMechanism from the URI to allow property-based credentials. If the URI already includes user info,
+            // keep the mechanism so the URI remains valid; property credentials will override later.
+            final boolean hasUserInfoInUri = uri.matches("(?i)^mongodb(?:\\+srv)?://[^/]*@.*");
```

Comment:
Similar to the `AUTH_MECHANISM_PATTERN`, recommend declaring this pattern statically and then matching.